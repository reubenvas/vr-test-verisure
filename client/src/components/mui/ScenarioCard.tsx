import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Grid from '@material-ui/core/Grid';
import Divider from '@material-ui/core/Divider';
import { Scenario } from 'backstopjs';
import { ExpansionPanel, ExpansionPanelSummary, ExpansionPanelDetails } from '@material-ui/core';
import sccenarioStore from '../../store/scenarioStore';
import ListItemTextInput from './ListItemTextInput';


const useStyles = makeStyles((theme) => createStyles({
    root: {
        minWidth: 800,
    },

    title: {
        fontSize: 14,
    },
    pos: {
        marginBottom: 12,
    },
    dividerFullWidth: {
        marginRight: 10,
    },
    listRoot: {
        width: '100%',
        maxWidth: 360,
        backgroundColor: theme.palette.background.paper,
    },
    noTopPadding: {
        paddingTop: 0,
    },
    noBottomPadding: {
        paddingBottom: 0,
    },
    textInMiddle: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
    },
}));

type propTypes = Scenario & {
    label: string;
    refUrl?: string;
    url: string;
    id: number;
};

const ScenarioCard = ({
    id, label, refUrl, url, ...restProps
}: propTypes): React.ReactElement => {
    const [expanded, setExpanded] = React.useState<boolean>(false);
    const classes = useStyles();

    const deleteScenario = (): void => {
        sccenarioStore.removeScenario(id);
    };

    const toggleExpandedView = (): void => {
        setExpanded(!expanded);
    };

    return (
        <Card className={classes.root}>
            <ExpansionPanel expanded={expanded}>
                <ExpansionPanelSummary>
                    <CardContent className={classes.noBottomPadding}>
                        <Grid container >
                            <Grid className={classes.textInMiddle} item xs={6}>
                                <Typography variant="h6" component="h2">
                                    {label}
                                </Typography>
                                <Typography
                                    className={classes.dividerFullWidth}
                                    color="textSecondary"
                                    display="block"
                                    variant="caption"
                                >
                                    Scenario label
                                </Typography>
                            </Grid>
                            <Grid item xs={6}>
                                <List className={classes.listRoot}>
                                    <ListItem className={`${classes.noTopPadding} ${classes.noBottomPadding}`}>
                                        <ListItemText primary={url} secondary="Url" />
                                    </ListItem>
                                    <Divider component="li" />
                                    <ListItem className={`${classes.noTopPadding} ${classes.noBottomPadding}`}>
                                        <ListItemText primary={refUrl} secondary="Reference url" />
                                    </ListItem>
                                </List>
                            </Grid>
                        </Grid>
                    </CardContent>
                </ExpansionPanelSummary>
                <ExpansionPanelDetails>
                    <List className={classes.listRoot}>
                        {Object.keys(restProps).map((key, i) => (
                            <div key={i}>
                                <ListItem className={`${classes.noTopPadding} ${classes.noBottomPadding}`} >
                                    <ListItemTextInput
                                        type={typeof restProps[key]}
                                        id={id} initialPrimary={restProps[key] === '' ? 'Not specified' : restProps[key]}
                                        secondary={key}
                                    />
                                </ListItem>
                                <Divider component="li" />
                            </div>
                        ))}
                    </List>

                </ExpansionPanelDetails>
            </ExpansionPanel>
            <CardActions>
                <Grid container justify="space-between" alignItems="baseline">
                    <Button size="medium" onClick={toggleExpandedView}>View {expanded ? 'less' : 'more'}</Button>
                    <Button size="small" onClick={deleteScenario}>Delete</Button>
                </Grid>
            </CardActions>
        </Card>
    );
};

export default ScenarioCard;
