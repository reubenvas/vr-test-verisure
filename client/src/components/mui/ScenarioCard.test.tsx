import React from 'react';
import renderer from 'react-test-renderer';
import ScenarioCard from './ScenarioCard';

describe('ScenarioCard component', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<ScenarioCard id={1} label='Test scenario card' url='https://test.com' refUrl='https://test.com' hideSelector='.test1' clickSelector='.test2' />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
