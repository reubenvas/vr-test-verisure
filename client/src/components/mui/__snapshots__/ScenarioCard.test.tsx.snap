// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioCard component snapshot renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiPaper-root MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content"
      >
        <div
          className="MuiCardContent-root makeStyles-noBottomPadding-7"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiGrid-root makeStyles-textInMiddle-8 MuiGrid-item MuiGrid-grid-xs-6"
            >
              <h2
                className="MuiTypography-root MuiTypography-h6"
              >
                Test scenario card
              </h2>
              <span
                className="MuiTypography-root makeStyles-dividerFullWidth-4 MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Scenario label
              </span>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
            >
              <ul
                className="MuiList-root makeStyles-listRoot-5 MuiList-padding"
              >
                <li
                  className="MuiListItem-root makeStyles-noTopPadding-6 makeStyles-noBottomPadding-7 MuiListItem-gutters"
                  disabled={false}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      https://test.com
                    </span>
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      Url
                    </p>
                  </div>
                </li>
                <li
                  className="MuiDivider-root"
                  role="separator"
                />
                <li
                  className="MuiListItem-root makeStyles-noTopPadding-6 makeStyles-noBottomPadding-7 MuiListItem-gutters"
                  disabled={false}
                >
                  <div
                    className="MuiListItemText-root MuiListItemText-multiline"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                    >
                      https://test.com
                    </span>
                    <p
                      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                    >
                      Reference url
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <ul
                className="MuiList-root makeStyles-listRoot-5 MuiList-padding"
              >
                <div>
                  <li
                    className="MuiListItem-root makeStyles-noTopPadding-6 makeStyles-noBottomPadding-7 MuiListItem-gutters"
                    disabled={false}
                  >
                    <div
                      className="MuiListItemText-root MuiListItemText-multiline"
                      onDoubleClick={[Function]}
                    >
                      <span
                        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                      >
                        .test1
                      </span>
                      <p
                        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                      >
                        hideSelector
                      </p>
                    </div>
                  </li>
                  <li
                    className="MuiDivider-root"
                    role="separator"
                  />
                </div>
                <div>
                  <li
                    className="MuiListItem-root makeStyles-noTopPadding-6 makeStyles-noBottomPadding-7 MuiListItem-gutters"
                    disabled={false}
                  >
                    <div
                      className="MuiListItemText-root MuiListItemText-multiline"
                      onDoubleClick={[Function]}
                    >
                      <span
                        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                      >
                        .test2
                      </span>
                      <p
                        className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary"
                      >
                        clickSelector
                      </p>
                    </div>
                  </li>
                  <li
                    className="MuiDivider-root"
                    role="separator"
                  />
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-baseline MuiGrid-justify-xs-space-between"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          View 
          more
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</div>
`;
